### PR TITLE
feat(function): add support for namespace

### PR DESCRIPTION
**Pull request by remyleone**

### CODE REVIEW

File: scaleway/resource_function_namespace.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+		UpdateContext: resourceScalewayFunctionNamespaceUpdate,
+		DeleteContext: resourceScalewayFunctionNamespaceDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultFunctionNamespaceTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
```
Comment (Monitob) [inline scaleway/resource_function_namespace.go:25]: What about the `secret_environment_variables` [attribute](https://github.com/scaleway/scaleway-sdk-go/blob/810e5a35c5722efd2b28bd967efaeec4651bd5d0/api/function/v1beta1/function_sdk.go#L798)

File: scaleway/resource_function_namespace.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+		return diag.FromErr(err)
+	}
+
+	ns, err := api.GetNamespace(&function.GetNamespaceRequest{
+		Region:      region,
+		NamespaceID: id,
+	}, scw.WithContext(ctx))
+
+	if err != nil {
+		if is404Error(err) {
```
Comment (Monitob) [inline scaleway/resource_function_namespace.go:97]: Why is this necessary?

File: scaleway/resource_function_namespace.go [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+	return resourceScalewayFunctionNamespaceRead(ctx, d, meta)
+}
+
+func resourceScalewayFunctionNamespaceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, id, err := functionAPIWithRegionAndID(meta, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ns, err := api.GetNamespace(&function.GetNamespaceRequest{
```
Comment (Monitob) [inline scaleway/resource_function_namespace.go:91]: Please use [this](https://github.com/scaleway/scaleway-sdk-go/blob/master/api/function/v1beta1/function_helpers.go#L27) instead

File: scaleway/resource_function_namespace.go [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"registry_endpoint": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The endpoint reachable by docker",
+			},
+			"registry_namespace_id": {
```
Comment (Monitob) [inline scaleway/resource_function_namespace.go:51]: Should we consider the functions and triggers in this PR? 🤔  and maybe change the resource in only resource_function. (open question)

File: scaleway/resource_function_namespace.go [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+
+	return nil
+}
+
+func resourceScalewayFunctionNamespaceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, id, err := functionAPIWithRegionAndID(meta, d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
```
Comment (Monitob) [inline scaleway/resource_function_namespace.go:121]: Please check the Namespace state before update

File: scaleway/resource_function_namespace.go [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+	}
+}
+
+func resourceScalewayFunctionNamespaceCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api, region, err := functionAPIWithRegion(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	ns, err := api.CreateNamespace(&function.CreateNamespaceRequest{
```
Comment (Monitob) [inline scaleway/resource_function_namespace.go:69]: Please check if namespace already exists and return a warning. You will find an example [here](https://github.com/scaleway/terraform-provider-scaleway/blob/master/scaleway/resource_domain_zone.go#L90)

File: scaleway/resource_function_namespace.go [hunk 7]
```diff
@@ -0,0 +1,160 @@
...
+				ForceNew:    true,
+				Optional:    true,
+				Description: "The name of the function namespace",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The description of the function namespace",
+			},
+			"environment_variables": {
```
[the remark this answers is not in the document]
Comment (Monitob) [reply scaleway/resource_function_namespace.go:38]: There are also 1000 chars limitations. more about limitations [about](https://www.scaleway.com/en/docs/compute/functions/reference-content/functions-limitations/).
Maybe only check the size of the collection is good enough

File: scaleway/resource_function_namespace.go (hunk 4 above)
Comment (remyleone) [reply scaleway/resource_function_namespace.go:54]: This feature will be handled in the function resource.

File: scaleway/resource_function_namespace.go (hunk 1 above)
Comment (remyleone) [reply scaleway/resource_function_namespace.go:26]: At the moment the way secret_environment_variables is implemented is a bit weird. I would like to talk a bit with the product team before implementing it.

File: docs/data-sources/function_namespace.md [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+```
+
+## Argument Reference
+
+- `name` - (Optional) The namespace name.
+  Only one of `name` and `namespace_id` should be specified.
+
+- `namespace_id` - (Optional) The namespace id.
+  Only one of `name` and `namespace_id` should be specified.
+-
```
Comment (Monitob) [inline docs/data-sources/function_namespace.md:32]: Please check trailing space here https://github.com/scaleway/terraform-provider-scaleway/runs/5305396640?check_suite_focus=true

**Review by Monitob — CHANGES_REQUESTED** [review]

**Review by Monitob — APPROVED** [review]
